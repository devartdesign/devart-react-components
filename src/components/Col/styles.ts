import { StyleSheet } from 'aphrodite/no-important';

export default StyleSheet.create({
  container: {
    position: 'relative',
    width: '100%',
    paddingRight: 15,
    paddingLeft: 15,
    boxSizing: 'border-box'
  }
});
