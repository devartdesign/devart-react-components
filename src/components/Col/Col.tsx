import React, { memo, CSSProperties } from 'react';
import { css } from 'aphrodite/no-important';

import styles from './styles';

export interface IColProps {
  className?: string;
  size?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
  style?: CSSProperties;
}

const Col = ({ children, className = '', size = 12, style = {} }: React.PropsWithChildren<IColProps>) => {
  return (
    <div
      key="Col"
      className={`${css(styles.container)} ${className}`.trim()}
      style={{ maxWidth: `${(size / 12) * 100}%`, flex: `0 0 ${(size / 12) * 100}%`, ...style }}
    >
      {children}
    </div>
  );
};

export default memo(Col);
