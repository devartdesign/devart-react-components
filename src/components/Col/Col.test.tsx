import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';

import Col from './Col';

describe('Col Component', () => {
  global.console.error = () => null;
  let Component;
  it('should render', () => {
    Component = mount(<Col>data</Col>);
    expect(create(Component).toJSON()).toMatchSnapshot();
  });

  it('should render with extra css classes', () => {
    Component = mount(
      <Col size={6} className="another-css-class">
        <p>data</p>
      </Col>
    );
    expect(create(Component).toJSON()).toMatchSnapshot();
  });
});
