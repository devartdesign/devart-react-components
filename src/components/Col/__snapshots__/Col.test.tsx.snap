// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col Component should render 1`] = `
.container {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

<div
  className="container"
  style={
    Object {
      "flex": "0 0 100%",
      "maxWidth": "100%",
    }
  }
>
  data
</div>
`;

exports[`Col Component should render with extra css classes 1`] = `
.container {
  position: relative;
  width: 100%;
  padding-right: 15px;
  padding-left: 15px;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

<div
  className="container another-css-class"
  style={
    Object {
      "flex": "0 0 50%",
      "maxWidth": "50%",
    }
  }
>
  <p>
    data
  </p>
</div>
`;
